sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp

```diff
@@ -37,16 +38,37 @@ namespace Azure { namespace Core { namespace Http {
      *
      * @remark Libcurl does revocation list check by default for SSL backends that supports this
      * feature. However, the Azure SDK overrides libcurl's behavior and disables the revocation list
-     * check by default.
-     *
+     * check by default. This ensures that the LibCURL behavior matches the WinHTTP behavior.
      */
     bool EnableCertificateRevocationListCheck = false;
+
+    /**
+     * @brief This option allows SSL connections to proceed even if there is an error retrieving the
+     * Certificate Revocation List.
+     *
+     * @remark Note that this only works when LibCURL is configured to use openssl as its TLS
+     * provider. That functionally limits this check to Linux only, and then only when openssl is
+     * configured (the default).
+     */
+    bool AllowFailedCrlRetrieval = false;
+
+    /**
+     * @brief A set of PEM encoded X.509 certificates and CRLs describing the certificates used to
+     * validate the server.
+     *
+     * @remark The Azure SDK will not directly validate these certificates.
+     *
+     * @remark More about this option:
+     * https://curl.haxx.se/libcurl/c/CURLOPT_CAINFO_BLOB.html
+     *
+     */
+    std::string PemEncodedExpectedRootCertificates;
   };
 
   /**
    * @brief Set the libcurl connection options like a proxy and CA path.
    */
-  struct CurlTransportOptions final
```

Comment:
Because `CurlWebSocketOptions` options derives from `CurlTransportOptions`. This is a tiny bit of websockets that creeps into this pull request, basically it came along for the ride. But I'd rather keep it in because it reduces the diffs with websockets.